internal/backup_util.go

```diff
@@ -199,13 +199,28 @@ func FolderSize(folder storage.Folder, path string) (int64, error) {
 // SplitPurgingBackups partitions backups to delete and retain, if no retains policy than retain all backups
 func SplitPurgingBackups(backups []TimedBackup,
 	retainCount *int,
-	retainAfter *time.Time) (purge, retain map[string]bool, err error) {
+	retainAfter *time.Time,
+	target *string,
+) (purge, retain map[string]bool, err error) {
 	retain = make(map[string]bool)
 	purge = make(map[string]bool)
-	retainAll := retainCount == nil && retainAfter == nil
+	retainAll := retainCount == nil && retainAfter == nil && target == nil
 	retainedCount := 0
+	deleteTarget := target != nil
 	for i := range backups {
 		backup := backups[i]
+
+		if deleteTarget {
+			if backup.Name() == *target {
+				purge[backup.Name()] = true
```

Comment:
I think that we shouldn't delete `target` backup if it is `permanent`.

How it works in Pg/MySQL:
https://github.com/wal-g/wal-g/blob/v3.0.1/internal/delete_handler.go#L391-L397

How it is documented in README.md:
`
Is used to delete backups and WALs before them. By default, ``delete`` will perform a dry run. If you want to execute deletion, you have to add ``--confirm`` flag at the end of the command. Backups marked as permanent will not be deleted.
`

Reply:
I've fixed it and added the test.